packages/webpack-cli/lib/webpack-cli.js

```diff
@@ -253,6 +253,14 @@ class WebpackCLI extends GroupHelper {
         let options = this.compilerConfiguration;
         let outputOptions = this.outputConfiguration;
 
+        let isWatchMode;
+
+        if (Array.isArray(options)) {
+            isWatchMode = options.some((options) => options.watch);
+        } else {
+            isWatchMode = options.watch;
+        }
```

Comment:
Why we need move it above?

Reply:
To do don't evaluate it on every call of callback as it will not change. 

Best would be to move the whole https://github.com/webpack/webpack-cli/blob/fd846e20d1b6c287570db2ce206d3756863ebc42/packages/webpack-cli/lib/webpack-cli.js#L331-L333
to plugin... We want to log the `Watching files...` after stats, so after plugin hooks. How about `process.nextTick` in this case 🤔 

```js
compiler.hooks.done.tap(PluginName, (compilation) => {
  logger.success(`Compilation${compilation.name ? `${compilation.name}` : ''} finished`);

  process.nextTick(() => {
      if (compiler.watchMode) {
          logger.success('Watching files for updates...');
      }
  });
});
```

Reply:
Sounds good